File: src/main/java/org/fountainmc/api/world/block/plants/Cactus.java
package org.fountainmc.api.world.block.plants;

import javax.annotation.Nonnegative;

public interface Cactus extends GrowingPlant {

    int MAXIMUM_GROWTH = 16;

    @Override
    Cactus withGrowth(@Nonnegative int growth);

    @Override
    default int getMaxGrowth() {
        return MAXIMUM_GROWTH;
    }

}

File: src/main/kotlin/org/fountainmc/api/world/block/plants/Cactus.kt
package org.fountainmc.api.world.block.plants

import org.fountainmc.api.BlockType

interface Cactus : GrowingPlant {
    override val blockType: BlockType<Cactus>

    override fun withGrowth(growth: Int): Cactus

    override val maxGrowth: Int
        get() = MAXIMUM_GROWTH

    companion object {
        val MAXIMUM_GROWTH = 16
    }

}
